### PR TITLE
New C++ implementation for Sleepy Cow Sort

### PR DESCRIPTION
**Pull request by DenialRiver1434**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/gold/usaco-898.mdx [hunk 1]
```diff
@@ -187,3 +187,54 @@ public class Sleepy {
...
+int N;
+vector<int> nums;
+
+int main() {
+	ifstream fin;
+	ofstream fout;
+	fin.open("sleepy.in");
+	fout.open("sleepy.out");
+
+	fin >> N;
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-898.mdx:216]: lowercase

File: solutions/gold/usaco-898.mdx [hunk 2]
```diff
@@ -187,3 +187,54 @@ public class Sleepy {
...
+
+Tree<int> sorted;
+int N;
+vector<int> nums;
+
+int main() {
+	ifstream fin;
+	ofstream fout;
+	fin.open("sleepy.in");
+	fout.open("sleepy.out");
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-898.mdx:214]:
```suggestion
	ifstream fin("sleepy.in");
	ofstream fout("sleepy.out");
```

Comment (SansPapyrus683) [reply solutions/gold/usaco-898.mdx:214]: also put the `ofstream` near the end where you actually use it

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/gold/usaco-898.mdx (hunk 2 above)
Comment (DenialRiver1434) [reply solutions/gold/usaco-898.mdx]: done with that

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**